src/shapes/Text/Text.ts

```diff
@@ -1492,12 +1529,13 @@ export class Text<
    * @param {String} property the property name
    * @returns the value of 'property'
    */
-  getValueOfPropertyAt(lineIndex: number, charIndex: number, property: string) {
+  getValueOfPropertyAt<T extends StylePropertiesType>(
+    lineIndex: number,
+    charIndex: number,
+    property: StylePropertiesType
+  ): this[T] {
```

Comment:
```suggestion
  getValueOfPropertyAt<T extends StylePropertiesType>(
    lineIndex: number,
    charIndex: number,
    property: T
  ): this[T] {
```

Seems this is what you want so TS infers it instead of you explicitly telling it.


![image](https://github.com/fabricjs/fabric.js/assets/34343793/e043e841-9c4a-41ae-be70-ec237ed57541)

![image](https://github.com/fabricjs/fabric.js/assets/34343793/b58df618-7e58-49e7-9c15-019519a261f2)

Autocomplete
![image](https://github.com/fabricjs/fabric.js/assets/34343793/3439e396-59c0-45ae-8327-d2d554d1a92e)

Reply:
i tried this as i stated before, it doesn't work.

Reply:
oh no what i tried was slightly different.
This works.
I didn't remove styleProperty from the argument i tried to use it.